### PR TITLE
MGMT-2752 - Add user-managed-networking to cluster (none-platform)

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
Generated config-yaml:
```

apiVersion: v1
baseDomain: redhat.com
networking:
  networkType: OpenShiftSDN
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  serviceNetwork:
  - 172.30.0.0/16
metadata:
  name: test-infra-cluster-assisted-installer
compute:
- hyperthreading: Enabled
  name: worker
  replicas: 0
controlPlane:
  hyperthreading: Enabled
  name: master
  replicas: 3
platform:
  none: {}
fips: false
pullSecret: 'PULL_SECRET'
sshKey: ssh-rsa SSH_KEY
```

**Review by avishayt — CHANGES_REQUESTED** [review]
So far it looks good, but if user-managed-networking==true, then we need to disable some validations.  For example we no longer need VIPs as those should be configured on the user's load balancer, and no longer need L2 connectivity because no keepalived.  @ori-amizur @romfreiman Anything else?

@yevgeny-shnaidman Does this affect day-2?

**Comment by yevgeny-shnaidman** [comment]
@avishayt yes, it affects. for day2 on ocp we assume BM config and extract cluster parameters from the configmap map in OCP.  We will need to update that code in order not to crush and update instruction manager for day2 ocp accordingly.
for day2 on cloud i think that no changes are needed

**Comment by avishayt** [comment]
> @avishayt yes, it affects. for day2 on ocp we assume BM config and extract cluster parameters from the configmap map in OCP. We will need to update that code in order not to crush and update instruction manager for day2 ocp accordingly.
> for day2 on cloud i think that no changes are needed

So when creating the add_hosts_cluster, they user will now provide the api_vip_dnsname according to whatever infra he configured, and the validations will remain the same?

**Comment by yevgeny-shnaidman** [comment]
> > @avishayt yes, it affects. for day2 on ocp we assume BM config and extract cluster parameters from the configmap map in OCP. We will need to update that code in order not to crush and update instruction manager for day2 ocp accordingly.
> > for day2 on cloud i think that no changes are needed
> 
> So when creating the add_hosts_cluster, they user will now provide the api_vip_dnsname according to whatever infra he configured, and the validations will remain the same?

for day2 on cloud we derive api_vip_dnsname from OCM, and use it for ignition pull validation, so i think nothing needs to be changed

File: internal/installcfg/installcfg.go [hunk 1]
```diff
@@ -30,7 +30,11 @@ type baremetal struct {
 }
 
 type platform struct {
-	Baremetal baremetal `yaml:"baremetal"`
+	Baremetal *baremetal    `yaml:"baremetal,omitempty"`
```
Comment (eranco74) [inline internal/installcfg/installcfg.go:33]: Nice

**Review by eranco74 — COMMENTED** [review]
Nice work
/lgtm
/approve

**Pull request merged**